### ISSUE TITLE
[Tabs] inconsistent keyboard navigation

### ISSUE THREAD

**Issue by kathydlavoie-ps** [chakra-ui/ark#3598]
### Description

When I navigate the Tabs component by keyboard, I expected focusable elements to consistently receive focus when tab is pressed from the Tab Trigger, but the TabPanel intermittently receives focus first.

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/p/devbox/l863rz

### Steps to Reproduce

1. Go to https://codesandbox.io/p/devbox/l863rz
2. Focus on first “React” Tab Trigger
3. Press Tab key to move focus to “React Button” in TabPanel
4. Press Shift + Tab to return to React Tab Trigger
5. Press right arrow to navigate to “Vue” trigger
6. Press Tab key to move focus to “Vue Button” in TabPanel
7. Repeat back and forth through Tab keyboard navigation - sometimes the button will be focused first, sometimes the TabPanel will be focused first

### Ark UI Version

5.20.0

### Framework

- [x] React
- [ ] Solid
- [ ] Vue
- [ ] Svelte

### Browser

Chrome

### Additional Information

Sometimes the button in the TabPanel receives focus first, sometimes the TabPanel receives focus first as an extra step.

[In the Tabs WAI-ARIA Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/), TabPanel should receive focus if there is not a focusable element, but if there is, the focusable element should receive initial focus on tab key press. This generally works, but sometimes does not work as expected with keyboard navigation.

https://github.com/user-attachments/assets/2bc55388-8b9c-402a-bfcc-48156846e819

**Comment by segunadebayo** [comment]
This is a good catch @kathydlavoie-ps. Thanks for reporting it.

I'm aware of your work on the PluralSight Panda design system 😄.

Found the bug and pushed a fix for it. Will do a new release shortly
